src/GraphUtils.ts

```diff
@@ -426,16 +430,31 @@ export function findTurnPts(pts: Point[], mode: string, isClosed: boolean = fals
         if (pts[i].y > pts[mod(i-1, pts.length)].y) {
             while (pts[i].y == pts[mod(i+1, pts.length)].y) {
                 i += 1;
+                if (i == pts.length - 1) {
+                    return false;
+                }
             }
             return pts[i].y > pts[mod(i+1, pts.length)].y;
         }
         return false;
     };
 
+    // does any point within 5 points of i have a y value greater than (maxima) / less than (minima) (remember y is inverted) EXTREMA_DETECTION_HEIGHT units further than pts[i].y?
```

Comment:
Slightly hard to parse
```suggestion
    // do any of the surrounding 5 points either side of i have a y value that is EXTREMA_DETECTION_HEIGHT units greater than (maxima) / less than (minima) pts[i].y? (remember y is inverted)
```